language/move-prover/src/prelude.bpl

```diff
@@ -928,18 +948,60 @@ procedure {:inline 1} $Vector_set(ta: TypeValue, src: Reference, i: Value, e: Va
     call WriteRef(src, v);
 }
 
-// TODO: implement the below vector methods
+procedure {:inline 1} $Vector_remove(ta: TypeValue, r: Reference, i: Value) returns (e: Value) {
+    var i_ind: int;
+    var v: Value;
+    var len: int;
 
-procedure {:inline 1} $Vector_contains(ta: TypeValue, vr: Reference, er: Reference) returns (res: Value)  {
-    res := DefaultValue;
-    assert false; // $Vector_contains not implemented
+    assume is#Integer(i);
+    i_ind := i#Integer(i);
+
+    v := $Dereference($m, r);
+    assume is#Vector(v);
+    assume $IsValidReferenceParameter($m, $local_counter, r);
+    len := $vlen(v);
+    if (i_ind < 0 || i_ind >= len) {
+        $abort_flag := true;
```

Comment:
Seems we are doing the range check here, so don't need the out-of-range behavior in $remove_vector.